packages/devtools_app/lib/src/screens/memory/shared/heap/model.dart

```diff
@@ -274,19 +274,23 @@ class SnapshotTaker {
   Future<AdaptedHeapData?> take() async {
     final snapshot = await snapshotMemory();
     if (snapshot == null) return null;
-    late final AdaptedHeapData result;
-    ga.timeSync(
-      analytics_constants.memory,
-      analytics_constants.MemoryTimeAnalytics.adaptSnapshot,
-      syncOperation: () => result = AdaptedHeapData.fromHeapSnapshot(snapshot),
-      screenMetricsProvider: () => _SnapshotAnalyticsMetrics(
-        numberOfObjects: snapshot.objects.length,
-      ),
-    );
-    return result;
+    return _adaptSnapshotGaWrapper(snapshot);
   }
 }
 
+AdaptedHeapData _adaptSnapshotGaWrapper(HeapSnapshotGraph graph) {
+  late final AdaptedHeapData result;
+  ga.timeSync(
+    analytics_constants.memory,
+    analytics_constants.MemoryTime.adaptSnapshot,
+    syncOperation: () => result = AdaptedHeapData.fromHeapSnapshot(graph),
+    screenMetricsProvider: () => _SnapshotAnalyticsMetrics(
```

Comment:
same comment as before about _SnapshotAnalyticsMetrics (we should create MemoryScreenMetrics to hold all our custom metrics, and then we need to add handling in _analytics_web.dart)